bundles/pcmumlclass/tools.vitruv.applications.pcmumlclass/src/tools/vitruv/applications/pcmumlclass/uml/UmlRepositoryAndSystemPackage.reactions

```diff
@@ -153,12 +154,36 @@ routine createCorrespondingRepository(uml::Package umlPkg, uml::Package umlParen
 	}
 }
 
+routine createOrFindCorrespondingSystem(uml::Package umlPkg, uml::Package umlParentPkg) {
+    match {
+        require absence of pcm::System corresponding to umlPkg tagged with TagLiterals.SYSTEM__SYSTEM_PACKAGE
+        val foundSystem = retrieve optional pcm::System corresponding to SystemPackage.Literals.SYSTEM
+            with foundSystem.entityName.toFirstLower == umlPkg.name // PCM system can be both upper and lower case
+    }
+    action {
+        call {
+            if (foundSystem.isPresent) {
+                addSystemCorrespondence(foundSystem.get, umlPkg)
+            } else {
+                createCorrespondingSystem(umlPkg, umlParentPkg)
+            }
+        }
+    }
+}
+
+routine addSystemCorrespondence(pcm::System pcmSystem, uml::Package umlPkg) {
+    action {
+        add correspondence between pcmSystem and umlPkg tagged with TagLiterals.SYSTEM__SYSTEM_PACKAGE
+    }
+}
+
 routine createCorrespondingSystem(uml::Package umlPkg, uml::Package umlParentPkg) {
 	action {
 		val pcmSystem = create pcm::System and initialize {
 			pcmSystem.entityName = umlPkg.name?.toFirstUpper
 		}
```

Comment:
The UML-to-PCM reactions create systems with a first letter upper case name. The Java-to-PCM reactions create systems with the name as it is. While this is inconsistent, it works with the where the find-or-create-pattern implementation that looks for systems with upper AND lower case names when trying to locate existing systems.

If we want to enforce first letter upper case system naming, I could change the Java-to-PCM system creation and change names of existing systems after being found. What do you think?

Reply:
I think the current implementation is absolutely fine. It actually means that both transformations consider systems and their package representation as consistent if the names are equal, independent of the capitalization of the first letter of the system name. The fact that the transformations enforce consistency with different patterns regarding the capitalization of the name do not conflict the underlying rules for which models are considered consistent.